src/Repository/AvailabilityQueryTrait.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repository;
+
+use App\Entity\AvailabilityInterface;
+use Doctrine\ORM\QueryBuilder;
+
+trait AvailabilityQueryTrait
+{
+    private function isEven(int $number): bool
+    {
+        return 0 == $number % 2;
+    }
+
+    private function addAvailabilityBetween(QueryBuilder $qb, \DateTimeImmutable $start, \DateTimeImmutable $end, string $availabilityClass, string $mainAlias, string $groupByField)
```

Comment:
This method can be a great addition to AvailabilitableRepositoryInterface too

Reply:
But i don't use it in the AvailabilitableRepository but their "parent" one